src/videojs-contrib-hls.js

```diff
@@ -436,20 +431,26 @@ export default class HlsHandler extends Component {
 const HlsSourceHandler = function(mode) {
   return {
     canHandleSource(srcObj) {
```

Comment:
@gkatsev it seems like the SourceHandler options should be passed to this method as well. As it stands, we can't figure out a way to force HLS to use Flash (or MSE) at the player level. Instead we have to use the kind of ugly global method you can see below.